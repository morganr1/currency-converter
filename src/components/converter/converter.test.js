import React from 'react';
import Converter from './converter.web';
import { shallow } from 'enzyme';


describe('Converter Component', () => {
    const mockProps = {
        themeType: 'light'
    };
    const mockSuccessResponse = {
        rates: {
            'EUR': 1,
            'GBP': 1.12,
            'USD': .89
        }
    };
    const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
    const mockFetchPromise = Promise.resolve({ // 3
        json: () => mockJsonPromise,
    });
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); // 4
    const wrapper = shallow(<Converter{...mockProps} />);

    it('renders correctly after fetching currencies from the api successfully', () => {
        expect(wrapper).toMatchSnapshot();
        expect(global.fetch).toHaveBeenCalledTimes(1);
        expect(global.fetch).toHaveBeenCalledWith('https://api.exchangeratesapi.io/latest');
    });
});
