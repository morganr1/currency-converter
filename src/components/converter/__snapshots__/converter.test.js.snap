// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter Component renders correctly after fetching currencies from the api successfully 1`] = `
<WithStyles(ForwardRef(Paper))
  className="converter-container"
  elevation={21}
  style={
    Object {
      "paddingBottom": 30,
      "paddingTop": 20,
    }
  }
>
  <WithStyles(ForwardRef(Typography))
    className="title-text"
    component="h5"
    variant="h5"
  >
    CURRENCY CONVERTER
  </WithStyles(ForwardRef(Typography))>
  <Result
    result={1.12}
    themeType="light"
    toCurrency="GBP"
  />
  <Currencies
    currencies={
      Array [
        "EUR",
        "EUR",
        "GBP",
        "USD",
      ]
    }
    disabled={false}
    fromCurrency="EUR"
    handleCurrencySelect={[Function]}
    toCurrency="GBP"
  />
  <Amount
    amount={1}
    buttonText="Clear"
    disabled={false}
    resetForm={[Function]}
    setAmountCallback={[Function]}
  />
</WithStyles(ForwardRef(Paper))>
`;
